lib/stripe.rb

```diff
@@ -48,47 +50,41 @@
 module Stripe
   DEFAULT_CA_BUNDLE_PATH = __dir__ + "/data/ca-certificates.crt"
 
-  @app_info = nil
-
-  @api_base = "https://api.stripe.com"
-  @connect_base = "https://connect.stripe.com"
-  @uploads_base = "https://files.stripe.com"
-
-  @log_level = nil
-  @logger = nil
-
-  @proxy = nil
-
-  @max_network_retries = 0
-  @max_network_retry_delay = 2
-  @initial_network_retry_delay = 0.5
-
-  @ca_bundle_path = DEFAULT_CA_BUNDLE_PATH
-  @ca_store = nil
-  @verify_ssl_certs = true
+  # map to the same values as the standard library's logger
+  LEVEL_DEBUG = Logger::DEBUG
+  LEVEL_ERROR = Logger::ERROR
+  LEVEL_INFO = Logger::INFO
 
-  @open_timeout = 30
-  @read_timeout = 80
+  @app_info = nil
 
-  @enable_telemetry = true
+  @configuration = Stripe::StripeConfiguration.setup
 
   class << self
-    attr_accessor :api_key
-    attr_accessor :api_version
+    extend Forwardable
+
+    # User configurable options
+    def_delegators :@configuration, :api_key, :api_key=
+    def_delegators :@configuration, :api_version, :api_version=
+    def_delegators :@configuration, :stripe_account, :stripe_account=
+    def_delegators :@configuration, :api_base, :api_base=
+    def_delegators :@configuration, :uploads_base, :uploads_base=
+    def_delegators :@configuration, :connect_base, :connect_base=
+    def_delegators :@configuration, :open_timeout, :open_timeout=
+    def_delegators :@configuration, :read_timeout, :read_timeout=
+    def_delegators :@configuration, :proxy, :proxy=
+    def_delegators :@configuration, :verify_ssl_certs, :verify_ssl_certs=
+    def_delegators :@configuration, :ca_bundle_path, :ca_bundle_path=
+    def_delegators :@configuration, :log_level, :log_level=
+    def_delegators :@configuration, :logger, :logger=
+    def_delegators :@configuration, :max_network_retries, :max_network_retries=
+    def_delegators :@configuration, :enable_telemetry=, :enable_telemetry?
+
+    # Internal configuarations
```

Comment:
Tiny typo here: "configurations"

Reply:
Whoops! I really need to configure my editor to spell check comments 😅